src/UCTNode.cpp

```diff
@@ -215,27 +215,23 @@ void UCTNode::set_policy(float policy) {
     m_policy = policy;
 }
 
-float UCTNode::get_variance(float default_var) const {
-    return m_visits > 1 ? m_squared_diff / (m_visits - 1) : default_var;
-}
-
-float UCTNode::get_stddev(float default_stddev) const {
-    return m_visits > 1 ? std::sqrt(get_variance()) : default_stddev;
+float UCTNode::get_eval_variance(float default_var) const {
```

Comment:
Have you considered using a `std::optional` here?  That way you don't need the `default_var` parameter.  You can just use `.value_or(0)`, this would also allow you to distinguish between cases where the variance is exactly the same as the passed `default_var` vs when the variance is merely undefined and taking the `default_var`.

Reply:
I don't think the default is even used anywhere right now. `get_eval_variance` is only needed in `get_eval_lcb` that only calls it when node has at least two visits. Probably not much point in using optional when it's never needed.

Reply:
std::optional is C++17 btw so not usable for now.

Reply:
You can use a signaling NaN to let things blow up if it's accidentally used.

Reply:
Signaling NaN didn't seem to do anything. It seems that it would require changing how floating point exceptions are handled for it to raise exception. Didn't want to go messing with that.